app/questionnaire/questionnaire_schema.py

```diff
@@ -94,11 +96,71 @@ def __init__(
         self._populate_answer_dependencies()
         self._populate_when_rules_section_dependencies()
         self._populate_calculated_summary_section_dependencies()
+        self._populate_min_max_for_numeric_answers()
 
     @cached_property
     def answer_dependencies(self) -> ImmutableDict[str, set[AnswerDependent]]:
         return ImmutableDict(self._answer_dependencies_map)
 
+    @cached_property
+    def min_and_max_map(self) -> ImmutableDict[str, str | dict[str, str]]:
+        return ImmutableDict(self._min_and_max_map)
+
+    def _append_to_min_max_map(
+        self,
+        min_max: Any,
+        answer_id: str,
+        answers: list,
+        default_min_max: str,
+    ) -> None:
+        longest_string = None
+        for answer in answers:
+            value = answer.get(min_max, {}).get("value")
+
+            if isinstance(value, int):
+                if longest_string and len(str(value)) > len(longest_string):
+                    longest_string = str(value)
+                else:
+                    longest_string = str(answer.get(min_max, {}).get("value"))
+
+            elif isinstance(value, dict) and value:
+                if (
+                    value.get("source") == "answers"
+                    and value["identifier"] in self._min_and_max_map
+                ):
+                    if longest_string and len(
+                        str(self._min_and_max_map[value["identifier"]][min_max])
+                    ) > len(longest_string):
+                        longest_string = str(
+                            self._min_and_max_map[value["identifier"]][min_max]
+                        )
+                    else:
+                        longest_string = str(
```

Comment:
The code in the if/else here seems to be doing the same thing? Is the if/else needed?